src/api/Feed.js

```diff
@@ -95,6 +103,83 @@ const getItemWithPendingReply = <T: { replies?: Array<Comment> }>(item: T, reply
 };
 
+const parseFileActivitiesResponseForFeed = (response?: { entries: FileActivity[] }) => {
+    if (!response || !response.entries || !response.entries.length) {
+        return { entries: [] };
+    }
+
+    const data = response.entries;
+    const parsedData = [];
+
+    data.map(item => {
+        if (!item.source) {
+            return item;
+        }
+
+        const source = { ...item.source };
+
+        switch (item.activity_type) {
+            case FILE_ACTIVITY_TYPE_TASK: {
+                const taskItem = source[FILE_ACTIVITY_TYPE_TASK];
+                // UAA follows a lowercased enum naming convention, convert to uppercase to align with task api
+
+                if (taskItem.assigned_to?.entries) {
+                    const assignedToEntries = taskItem.assigned_to.entries.map(entry => {
+                        const assignedToEntry = { ...entry };
+
+                        assignedToEntry.role = entry.role.toUpperCase();
+                        assignedToEntry.status = entry.status.toUpperCase();
+
+                        return assignedToEntry;
+                    });
+                    // $FlowFixMe Converting to uppercase makes role and status a string, which is incompatible with string literal
+                    taskItem.assigned_to.entries = assignedToEntries;
+                }
+                if (taskItem.completion_rule && isString(taskItem.completion_rule)) {
+                    taskItem.completion_rule = taskItem.completion_rule.toUpperCase();
+                }
+                if (taskItem.status && isString(taskItem.status)) {
+                    taskItem.status = taskItem.status.toUpperCase();
+                }
+                if (taskItem.task_type && isString(taskItem.task_type)) {
+                    taskItem.task_type = taskItem.task_type.toUpperCase();
+                }
+                // $FlowFixMe File Activities only returns a created_by user, Flow type fix is needed
+                taskItem.created_by = {
+                    target: taskItem.created_by,
+                };
+                break;
+            }
+            case FILE_ACTIVITY_TYPE_COMMENT: {
+                const commentItem: Comment = source[FILE_ACTIVITY_TYPE_COMMENT];
+
+                if (commentItem.replies && commentItem.replies.length) {
+                    const replies = commentItem.replies.map(reply => {
+                        const commentItemReply = { ...reply };
+                        commentItemReply.tagged_message = reply.tagged_message || reply.message || '';
+
+                        return commentItemReply;
+                    });
+
+                    commentItem.replies = replies;
+                }
+
+                commentItem.tagged_message = commentItem.tagged_message || commentItem.message || '';
+                break;
+            }
+            default: {
+                break;
+            }
+        }
+
+        parsedData.push(...Object.values(source));
```

Comment:
Can we `return source` here or is the Object.values call needed?

Reply:
Can't easily return source here since the key name depends on the type.

Reply:
Actually, I think we can if we make a case for every type. Which might be better since it's more explicit.